android/sloth-pwhash-libsodium/src/main/java/com/lambdapioneer/sloth/pwhash_libsodium/LibSodiumArgon2PwHash.kt

```diff
@@ -18,25 +21,33 @@ class LibSodiumArgon2PwHash(
 
     override fun deriveHash(
         salt: ByteArray,
-        password: String,
+        password: CharArray,
         outputLengthInBytes: Int,
     ): ByteArray {
         require(salt.size == SALT_LEN)
         require(outputLengthInBytes > 0)
         require(password.isNotEmpty())
 
-        val passwordBytes = password.encodeToByteArray()
+        val passwordCharBuffer = CharBuffer.wrap(password)
+        val passwordBytes = Charsets.UTF_8.encode(passwordCharBuffer).array()
+
         val result = ByteArray(outputLengthInBytes)
 
-        val libSodiumResult = libSodium.crypto_pwhash(
-            result, result.size.toLong(),
-            passwordBytes, passwordBytes.size.toLong(),
-            salt,
-            params.opsLimit,
-            params.memLimit,
-            com.goterl.lazysodium.interfaces.PwHash.Alg.PWHASH_ALG_ARGON2ID13.value,
-        )
-        check(libSodiumResult == 0)
+        try {
+            val libSodiumResult = libSodium.crypto_pwhash(
+                /* outputHash = */ result,
+                /* outputHashLen = */ result.size.toLong(),
+                /* password = */ passwordBytes,
+                /* passwordLen = */ passwordBytes.size.toLong(),
+                /* salt = */ salt,
+                /* opsLimit = */ params.opsLimit,
+                /* memLimit = */ params.memLimit,
+                /* alg = */ LibSodiumPwHash.Alg.PWHASH_ALG_ARGON2ID13.value,
+            )
+            check(libSodiumResult == 0)
+        } finally {
+            Arrays.fill(passwordBytes, 0x00)
+        }
```

Comment:
Note for myself: this is currently yielding wrong results. We must reference into the byte array using the `limit` offset from the byte buffer.